### ISSUE TITLE
"Unable to build LLVM" error with 2023.1

### ISSUE THREAD

**Issue by MatthewFluet** [smlnj/smlnj#279]
### Version

2023.1 (latest)

### Operating System

- [ ] Any
- [X] Linux
- [ ] macOS
- [ ] Windows
- [ ] Other Unix

### OS Version

Fedora 38

### Processor

- [ ] Any
- [ ] Arm (using Rosetta)
- [X] x86-64 (64-bit)
- [ ] Other

### System Component

Installation

### Severity

Critical

### Description

Running the provided `build.sh` terminates with:

```
[ 11%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Signals.cpp.o
In file included from /home/matthew/devel/smlnj/smlnj-2023.1/llvm/src/lib/Support/Signals.cpp:14:
/home/matthew/devel/smlnj/smlnj-2023.1/llvm/src/include/llvm/Support/Signals.h:117:8: error: variable or field ‘CleanupOnSignal’ declared void
  117 |   void CleanupOnSignal(uintptr_t Context);
      |        ^~~~~~~~~~~~~~~
/home/matthew/devel/smlnj/smlnj-2023.1/llvm/src/include/llvm/Support/Signals.h:117:24: error: ‘uintptr_t’ was not declared in this scope
  117 |   void CleanupOnSignal(uintptr_t Context);
      |                        ^~~~~~~~~
/home/matthew/devel/smlnj/smlnj-2023.1/llvm/src/include/llvm/Support/Signals.h:18:1: note: ‘uintptr_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
   17 | #include <string>
  +++ |+#include <cstdint>
   18 | 
[ 11%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/TargetRegistry.cpp.o
[ 11%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/ThreadLocal.cpp.o
[ 11%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Threading.cpp.o
In file included from /home/matthew/devel/smlnj/smlnj-2023.1/llvm/src/lib/Support/Signals.cpp:220:
/home/matthew/devel/smlnj/smlnj-2023.1/llvm/src/lib/Support/Unix/Signals.inc:348:44: error: ‘void llvm::sys::CleanupOnSignal(uintptr_t)’ should have been declared inside ‘llvm::sys’
  348 | void sys::CleanupOnSignal(uintptr_t Context) {
      |                                            ^
[ 11%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Valgrind.cpp.o
[ 11%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Watchdog.cpp.o
make[2]: *** [lib/Support/CMakeFiles/LLVMSupport.dir/build.make:1742: lib/Support/CMakeFiles/LLVMSupport.dir/Signals.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:3088: lib/Support/CMakeFiles/LLVMSupport.dir/all] Error 2
make: *** [Makefile:156: all] Error 2

real    1m27.050s
user    5m14.736s
sys     0m17.362s
./build.sh: !!! Unable to build LLVM
```

I include the full build transcript below, but I didn't see any other meaningful warnings or errors from the `./configure` step of LLVM.

### Transcript

[smlnj-2023.1.build.txt](https://github.com/smlnj/smlnj/files/13676058/smlnj-2023.1.build.txt)

### Steps to Reproduce

```
$ git clone https://github.com/smlnj/smlnj.git
$ cd smlnj
$ curl -O https://smlnj.org/dist/working/2023.1/boot.amd64-unix.tgz
$ ./build.sh -verbose
```

### Email address

Matthew.Fluet@gmail.com

**Comment by ratmice** [comment]
@JohnReppy I think the `<something>` got swallowed as html in `#include <something>` to avoid that, try wrapping it in backticks

**Comment by JohnReppy** [comment]
Thanks for noticing that; I've fixed it.

**Comment by MatthewFluet** [comment]
I have `clang++` v16 installed.

I tried adding `#include <cstdint>` to `Signals.h` and the LLVM build is getting further.

**Comment by MatthewFluet** [comment]
With that change, the build did complete.

I get the `fgrep: warning: fgrep is obsolescent; using grep -F` warning that was reported in smlnj/legacy#286 and reported as fixed in smlnj/legacy@1346e96, which seems not to have been ported.

**Comment by JohnReppy** [comment]
Thanks for testing the patch.  I will modify the code with the extra `#include`.  I looked at more recent versions of **LLVM** and they have that extra include in the `Signals.h` header.
